canonical-web-and-design/canonicalwebteam.search
# v0.1.0: Create the extension from code in docs.snapcraft.io

The search module, to help with adding search to Flask sites. The code was basically copies from docs.snapcraft.io, with some small restructuring.

QA
--

You can see this in action using [a branch of maas.io I prepared](https://github.com/canonical-web-and-design/maas.io/pull/364):

First, get hold of the "Custom Search API key" from [the Google cloud API credentials page](https://console.cloud.google.com/apis/credentials?project=ubuntu-search-1530889417216) - if you don't have access to that page, ask @nottrobin or someone for access.

Then run from that branch:

``` bash
git clone -b search-module https://github.com/nottrobin/maas.io maas.io-search
cd maas.io-search
./run --env SEARCH_API_KEY={the-api-key}
```

Now go to http://0.0.0.0:8006/docs, search for something and see the search results at e.g. http://0.0.0.0:8006/docs/search?q=maas.

(Note: The styling will look whack - this is acceptable for now, this is just a proof of concept)
